lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -246,6 +239,20 @@ def __init__(self, *args, distro: Type[OpenSearchDistribution] = None):
         self.framework.observe(self.on.set_password_action, self._on_set_password_action)
         self.framework.observe(self.on.get_password_action, self._on_get_password_action)
 
+        self.cos_integration = COSAgentProvider(
+            self,
+            relation_name=COSRelationName,
+            metrics_endpoints=[],
+            scrape_configs=self._scrape_config,
+            refresh_events=[
```

Comment:
Noob question. you reference this change in your PR message:
> Fixes data node integration with COS: COSAgentProvider was not tracking peer-cluster-* events, which means it was not receiving changes done to the COSUser by the main orchestrator

What do you mean by " it was not receiving changes done to the COSUser by the main orchestrator" - what does this do to the dashboard data. Is it needed across all DPE charms?

Reply:
@MiaAltieri in large deployments, only the `main_orchestrator` changes the password of the system users. That means, user `monitor`, which we use to authenticate COS, will be defined by one of the juju apps involved. Now, `main_orchestrator` then publishes that change to other apps. That is why we listen to `peer-cluster-*`.